### PR TITLE
Check permissions

### PR DESCRIPTION
**Pull request by bugrevealingbme**
I don't know if I need to add it exactly like this, but I need it

### CODE REVIEW

File: screen_brightness_android/android/src/main/kotlin/com/aaassseee/screen_brightness_android/ScreenBrightnessAndroidPlugin.kt [hunk 1]
```diff
@@ -335,6 +336,18 @@ class ScreenBrightnessAndroidPlugin : FlutterPlugin, MethodCallHandler, Activity
         )
     }
 
+    private fun checkSystemCanWrite(result: MethodChannel.Result): Boolean {
```
Comment (aaassseee) [inline screen_brightness_android/android/src/main/kotlin/com/aaassseee/screen_brightness_android/ScreenBrightnessAndroidPlugin.kt:339]: should rename to handleIsSystemSettingPermissionGrantedMethodCall with Unit return type

File: screen_brightness_android/android/src/main/kotlin/com/aaassseee/screen_brightness_android/ScreenBrightnessAndroidPlugin.kt [hunk 2]
```diff
@@ -335,6 +336,18 @@ class ScreenBrightnessAndroidPlugin : FlutterPlugin, MethodCallHandler, Activity
...
+        val context = activity.applicationContext
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            val canWrite = Settings.System.canWrite(context)
+            result.success(canWrite)
+            canWrite
```
Comment (aaassseee) [inline screen_brightness_android/android/src/main/kotlin/com/aaassseee/screen_brightness_android/ScreenBrightnessAndroidPlugin.kt:344]: remove unused return

File: screen_brightness/pubspec.yaml [hunk 3]
```diff
@@ -91,4 +91,4 @@ flutter:
   #         weight: 700
   #
   # For details regarding fonts in packages, see
-  # https://flutter.dev/custom-fonts/#from-packages
```
Comment (aaassseee) [inline screen_brightness/pubspec.yaml:94]: revert incorrect formatting

File: screen_brightness/pubspec.yaml [hunk 4]
```diff
@@ -30,9 +30,9 @@ dev_dependencies:
   plugin_platform_interface: ">=2.0.0 <3.0.0"
   async: ">=2.0.0 <3.0.0"
 
-#dependency_overrides:
-#  screen_brightness_platform_interface:
-#    path: ../screen_brightness_platform_interface/
+dependency_overrides:
+  screen_brightness_platform_interface:
+    path: ../screen_brightness_platform_interface/
```
Comment (aaassseee) [inline screen_brightness/pubspec.yaml:35]: revert dependency_overrides when doing pull request

File: screen_brightness/lib/screen_brightness.dart [hunk 5]
```diff
@@ -45,6 +45,11 @@ class ScreenBrightness {
   /// Code: -9, Message: value returns null
   Future<double> get system => _platform.system;
 
+  /// Check modify system settings
+  /// returns true or false
+  Future<bool> checkSystemCanWrite() =>
```
Comment (aaassseee) [inline screen_brightness/lib/screen_brightness.dart:50]: should rename to isSystemSettingPermissionGranted()

**Comment by bugrevealingbme** [comment]
I could not test it locally. I would appreciate if you add the feature

**Comment by aaassseee** [comment]
> I could not test it locally. I would appreciate if you add the feature

It is ok, I will continue to develop after merging this pull request. Thanks for the effort though

**Comment by aaassseee** [comment]
Will do it in this weekend

**Review by aaassseee — APPROVED** [review]

**Comment by aaassseee** [comment]
@bugrevealingbme this pull request will be publish after physical iOS device is being tested

**Comment by bugrevealingbme** [comment]
Do you need extra permissions for this on iOS?

**Comment by aaassseee** [comment]
iOS does not have any system setting API for developers

**Comment by bugrevealingbme** [comment]
await ScreenBrightness.instance.canChangeSystemBrightness did not return true. I don't understand the problem

**Comment by aaassseee** [comment]
@bugrevealingbme what are you trying to achieve?

**Comment by aaassseee** [comment]
If you don't allow the permission it will always return false

**Comment by bugrevealingbme** [comment]
I know, I gave permission

**Comment by aaassseee** [comment]
Has the permission popped when adjusting system brightness?

**Comment by bugrevealingbme** [comment]
Yes

**Comment by aaassseee** [comment]
have you accepted the permission?

**Comment by bugrevealingbme** [comment]
[log] MissingPluginException(No implementation found for method canChangeSystemBrightness on channel github.com/aaassseee/screen_brightness)

**Comment by bugrevealingbme** [comment]
Using like this

dependency_overrides:
  screen_brightness:
    git:
      url: https://github.com/aaassseee/screen_brightness.git
      ref: develop
      path: screen_brightness
  screen_brightness_platform_interface:
    git:
      url: https://github.com/aaassseee/screen_brightness.git
      ref: develop
      path: screen_brightness_platform_interface

**Comment by bugrevealingbme** [comment]
Fixed with adding 💯 

 screen_brightness_android:
    git:
      url: https://github.com/aaassseee/screen_brightness.git
      ref: develop
      path: screen_brightness_android

**Pull request merged**